src/main/scala/org/broadinstitute/dsde/workbench/sam/google/GoogleExtensions.scala

```diff
@@ -645,6 +646,22 @@ class GoogleExtensions(
     )
   }
 
+  def getRequesterPaysSignedUrl(
+      samUser: SamUser,
+      bucket: GcsBucketName,
+      name: GcsBlobName,
+      duration: Option[Long],
+      requesterPaysProject: Option[GoogleProject],
+      samRequestContext: SamRequestContext
+  ): IO[URL] = {
+    val urlParamsMap: Map[String, String] = requesterPaysProject.map(p => Map(userProjectQueryParam -> p.value)).getOrElse(Map.empty)
+    for {
+      petKey <- IO.fromFuture(IO(getArbitraryPetServiceAccountKey(samUser, samRequestContext)))
```

Comment:
Why do this and not mimic the current behavior of specifying a service account in the URL?